Exception during compilation when using interop for dagger Multibinds

### Description

```
Compilation exited with INTERNAL_ERROR but expected OK:
e: java.lang.AssertionError: No such value argument slot in IrConstructorCallImpl: 0 (total=0)
Expression: CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in dagger.multibindings.Multibinds' type=dagger.multibindings.Multibinds origin=null
	at org.jetbrains.kotlin.ir.expressions.IrExpressionsKt.checkArgumentSlotAccess(IrExpressions.kt:78)
	at org.jetbrains.kotlin.ir.expressions.IrMemberAccessExpression.getValueArgument(IrMemberAccessExpression.kt:415)
	at dev.zacsweers.metro.compiler.ir.IrKt.getSingleConstBooleanArgumentOrNull(ir.kt:627)
	at dev.zacsweers.metro.compiler.ir.transformers.DependencyGraphTransformer.createBindingGraph(DependencyGraphTransformer.kt:849)
	at dev.zacsweers.metro.compiler.ir.transformers.DependencyGraphTransformer.getOrBuildDependencyGraph(DependencyGraphTransformer.kt:653)
	at dev.zacsweers.metro.compiler.ir.transformers.DependencyGraphTransformer.visitClass(DependencyGraphTransformer.kt:200)
	at dev.zacsweers.metro.compiler.ir.transformers.DependencyGraphTransformer.visitClass(DependencyGraphTransformer.kt:150)
```

### Self-contained Reproducer

```kotlin
@DependencyGraph
interface ExampleGraph {
   val intSet: Set<Int>
            
   @dagger.multibindings.Multibinds fun emptyFoo(): Set<Int>
}
```

### Metro version

0.1.3

### Context

_No response_

--- a/compiler/src/main/kotlin/dev/zacsweers/metro/compiler/ir/ir.kt
+++ b/compiler/src/main/kotlin/dev/zacsweers/metro/compiler/ir/ir.kt
@@ -623,8 +623,15 @@ internal fun IrClass.implementsAny(
   }
 }
 
-internal fun IrConstructorCall.getSingleConstBooleanArgumentOrNull(): Boolean? =
-  (getValueArgument(0) as IrConst?)?.value as Boolean?
+/**
+ * Returns the single const boolean argument of this constructor call or null if...
+ * - The number of arguments is not 1
+ * - The argument is not a const boolean
+ */
+internal fun IrConstructorCall.getSingleConstBooleanArgumentOrNull(): Boolean? {
+  if (valueArgumentsCount != 1) return null
+  return (getValueArgument(0) as IrConst?)?.value as Boolean?
+}
 
 internal fun IrConstructorCall.getConstBooleanArgumentOrNull(name: Name): Boolean? =
   (getValueArgument(name) as IrConst?)?.value as Boolean?

--- a/compiler/src/test/kotlin/dev/zacsweers/metro/compiler/transformers/DaggerInteropTest.kt
+++ b/compiler/src/test/kotlin/dev/zacsweers/metro/compiler/transformers/DaggerInteropTest.kt
@@ -3,6 +3,7 @@
 package dev.zacsweers.metro.compiler.transformers
 
 import com.google.common.truth.Truth.assertThat
+import com.tschuchort.compiletesting.KotlinCompilation
 import com.tschuchort.compiletesting.SourceFile
 import com.tschuchort.compiletesting.configureKsp
 import dagger.Lazy
@@ -10,6 +11,7 @@ import dagger.internal.codegen.KspComponentProcessor
 import dev.zacsweers.metro.compiler.ExampleGraph
 import dev.zacsweers.metro.compiler.MetroCompilerTest
 import dev.zacsweers.metro.compiler.MetroOptions
+import dev.zacsweers.metro.compiler.assertDiagnostics
 import dev.zacsweers.metro.compiler.callProperty
 import dev.zacsweers.metro.compiler.createGraphWithNoArgs
 import dev.zacsweers.metro.compiler.generatedMetroGraphClass
@@ -43,6 +45,7 @@ class DaggerInteropTest : MetroCompilerTest() {
           setOf(ClassId.fromString("dagger/assisted/AssistedFactory")),
         customAssistedInjectAnnotations =
           setOf(ClassId.fromString("dagger/assisted/AssistedInject")),
+        customMultibindsAnnotations = setOf(ClassId.fromString("dagger/multibindings/Multibinds")),
       )
 
   @Test
@@ -447,6 +450,32 @@ class DaggerInteropTest : MetroCompilerTest() {
       assertThat(fooInstance).isNotNull()
       assertThat(fooInstance).isInstanceOf(DaggerInteropDoubleCheck::class.java)
       assertThat(fooInstance.get().javaClass.name).isEqualTo("test.Foo")
+    }
+  }
+
+  // Regression test for https://github.com/ZacSweers/metro/issues/334
+  @Test
+  fun `multibinds are usable`() {
+    compile(
+      source(
+        """
+          @DependencyGraph
+          interface ExampleGraph {
+             @dagger.multibindings.Multibinds fun ints(): Set<Int>
+          }
+        """
+          .trimIndent()
+      ),
+      expectedExitCode = KotlinCompilation.ExitCode.COMPILATION_ERROR,
+    ) {
+      assertDiagnostics(
+        """
+          e: ExampleGraph.kt:6:1 [Metro/EmptyMultibinding] Multibinding 'kotlin.collections.Set<kotlin.Int>' was unexpectedly empty.
+
+          If you expect this multibinding to possibly be empty, annotate its declaration with `@Multibinds(allowEmpty = true)`.
+        """
+          .trimIndent()
+      )
     }
   }
 }

